docs/specs/user-export.md

```diff
@@ -0,0 +1,284 @@
+# User Export
+
+User Export allows the developer to bulk export all users from Authgear to a file.
+
+## About User Export
+
+- It is not a synchronous operation. The export is created and runs in the background. The developer can query the status of it.
+
+## Create an export
+
+The endpoint is `POST /_api/admin/users/export`.
+
+- The endpoint requires Admin API JWT token to access.
+- This endpoint is added to Admin API server.
+- It is not part of the GraphQL API.
+- At most 1 running export at any given moment for a project.
+- The rate limit of creating a export is 24 per 24h. This means failed requests do not count.
+- A pending export lasts for 24h before it expires.
+
+### The request body of Create an export
+
+```
+{
+  "format": "ndjson",
+  "csv": {
+    "fields": [
+      {
+        "json_pointer": "/sub",
+        "field_name": "user_id"
+      }
+    ]
+  }
+}
+```
+
+- `format`: Required. It must be `ndjson` or `csv`.
+  - `ndjson`: The output is a ndjson file. See https://github.com/ndjson/ndjson-spec
+  - `csv`: The output is a CSV file. See https://datatracker.ietf.org/doc/html/rfc4180
+- `csv.fields`: Required when `format` is `csv`. It must be an non-empty array.
```

Comment:
Shall we make it optional as well? So when it's optional, use one of the approaches in the below's naming way to export all fields?

Reply:
Agree that we should have a default format for csv too.